lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
+
+    var len = o.length >>> 0;
```

Comment:
Also, fromIndex | 0 check whether fromIndex if defined and set before assigning the value. If it is not, the default value will be 0 instead of undefined for example.